oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -78,15 +80,62 @@ protected void setProviders() throws IOException {
         this.indexerProviders = createProviders();
     }
 
+    private static class MongoNodeStateEntryTraverserFactory implements NodeStateEntryTraverserFactory {
+
+        private static final AtomicInteger traverserInstanceCounter = new AtomicInteger(0);
```

Comment:
It is a bit weird to use a static (global) variable... I wonder if it's needed? If this is to help debugging, then that's good enough. But if we want to keep it static, then I wonder if it should be all caps.

Reply:
Yes i actually added this for better debugging information in logs.
Regarding all caps, i think it is generally used for constants (static and final). While these two modifiers are used here, the value here is not constant (only the reference is). So i didn't capitalize the name :)